far/viewer.cpp

```diff
@@ -4089,3 +4159,80 @@ void Viewer::vgetc_cache::compact()
 	m_End = std::copy(m_Iterator, m_End, m_Buffer);
 	m_Iterator = m_Buffer;
 }
+
+#ifdef ENABLE_TESTS
+
+#include "testing.hpp"
+
+TEST_CASE("hex.mode.bytes.area.width")
+{
+	struct test_data
+	{
+		int Bytes;
+		int Width;
+	};
+
+	static const test_data TestDataPoints[] =
+	{
+		{  8,  24 }, // > 2D 2D 2D 2D 2D 2D 2D 2D<
+		{  9,  29 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D<
+		{ 10,  32 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D<
+		{ 13,  41 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D<
+		{ 15,  47 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D<
+		{ 16,  50 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D<
+		{ 17,  55 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D<
+		{ 20,  64 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D<
+		{ 24,  76 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D<
+		{ 25,  81 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D<
+		{ 30,  96 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D<
+		{ 32, 102 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D<
+		{ 33, 107 }, // > 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D 2D 2D 2D 2D 2D 2D 2D │ 2D<
+	};
+
+	for (const auto& TestDataPoint : TestDataPoints)
+	{
+		REQUIRE(TestDataPoint.Width == hex_mode_bytes_area_wdth(TestDataPoint.Bytes));
+	}
+}
+
+TEST_CASE("hex.mode.bytes.from.line.width")
```

Comment:
Please don't use `.` as a word separator: UI tools often group tests by these tokens, supposedly assuming something like a namespace.class.method structure, which doesn't make any sense in this case:

![image](https://github.com/FarGroup/FarManager/assets/11453922/76f3eecd-1739-4abb-bf94-913dcf3e525f)

Better something like "area.function_being_tested" e.g. "viewer.hex_mode_bytes_from_line_width"

Reply:
LOL. I've never given a thought to this naming. Makes sense, of course. Fixed.